kong/init.lua

```diff
@@ -667,78 +676,80 @@ function Kong.init_worker()
 
   kong.db:set_events_handler(worker_events)
 
-  if kong.configuration.database == "off" then
-    -- databases in LMDB need to be explicitly created, otherwise `get`
-    -- operations will return error instead of `nil`. This ensures the default
-    -- namespace always exists in the
-    local t = lmdb_txn.begin(1)
-    t:db_open(true)
-    ok, err = t:commit()
-    if not ok then
-      stash_init_worker_error("failed to create and open LMDB database: " .. err)
-      return
-    end
-
-    if not has_declarative_config(kong.configuration) and
-      declarative.get_current_hash() ~= nil then
-      -- if there is no declarative config set and a config is present in LMDB,
-      -- just build the router and plugins iterator
-      ngx_log(ngx_INFO, "found persisted lmdb config, loading...")
-      local ok, err = declarative_init_build()
+  if process.type ~= "privileged agent" then
```

Comment:
It is a huge `if-then-end` block, could we use a simple `goto` to keep the code clean?